backend/chainlit/server.py

```diff
@@ -870,13 +871,67 @@ async def upload_file(
     assert file.filename, "No filename for uploaded file"
     assert file.content_type, "No content type for uploaded file"
 
+    validate_file_upload(file)
+
     file_response = await session.persist_file(
         name=file.filename, content=content, mime=file.content_type
     )
 
     return JSONResponse(content=file_response)
 
 
+def validate_file_upload(file: UploadFile):
+    if config.features.spontaneous_file_upload is None:
+        return  # TODO: if it is not configured what should happen?
```

Comment:
The most elegant solution is to change the default value for `config.features.spontaneous_file_upload` to always be populated with `SpontaneousFileUploadFeature` then to have `enabled` default to `False` and move  `max_files` and `max_size_mb` [default values from the frontend](https://github.com/Chainlit/chainlit/blob/2ba2c59c241e1227a11a9b439dabba5806e049b5/libs/copilot/src/chat/body.tsx#L41) to the backend config. This would remove the `if ... is None` in favour of more explicit code. But I get it if perhaps that's out of scope and left for a later PR.

The second best would be to follow the [frontend](https://github.com/Chainlit/chainlit/blob/2ba2c59c241e1227a11a9b439dabba5806e049b5/libs/copilot/src/chat/body.tsx#L151) in this and not allow uploads unless explicitly enabled. Definitely, we should try to avoid having a TODO in a PR. ;)

Reply:
The current behaviour for the e2e tests is that it is None. So rejecting upload when it is not configured will break the tests and might also break things for users.

I agree that the best would be to not have None as the default, but this indeed seems out of scope.